src/main/daml/Daml/Finance/Instrument/Swap/Fpml/Util.daml

```diff
@@ -80,6 +80,35 @@ adjustDateAccordingToBusinessDayAdjustments unadjustedDate businessDayAdjustment
       NoAdjustment -> pure $ unadjustedDate
       _ -> getCalendarsAndAdjust unadjustedDate businessDayAdjustments issuer calendarDataAgency
 
+-- Adjust payment schedule according to paymentDaysOffset (if available).
+applyPaymentDaysOffset : [SchedulePeriod] -> PaymentDates -> [HolidayCalendarData] ->
+  [SchedulePeriod]
+applyPaymentDaysOffset paymentSchedule paymentDates paymentCalendars =
+  case paymentDates.paymentDaysOffset of
+    Some dateOffset -> case dateOffset.dayType of
+      Some Business -> map (\p -> SchedulePeriod with
```

Comment:
I think you can just write `\p -> p with adjustedEndDate = addBusinessDays cals nDays p.adjustedendDate`